### PR TITLE
[WIP] Crud_Mvc_Validator_Ajax

### PR DESCRIPTION
**Pull request by tran-sang-mulodo**

### CODE REVIEW

**Comment by tran-hai-mulodo** [comment]
@tran-sang-mulodo 

Please resolve conflict files then re-assign me again. Thanks.

**Comment by tran-hai-mulodo** [comment]
@vo-thanh-mulodo 
cc: @tran-sang-mulodo 

I reviewed the convention. Please check the functions to make sure it works well. Thanks.

File: app/controllers/authors_controller.rb [hunk 1]
```diff
@@ -15,33 +15,31 @@ def show
 
   def new
     @authors = Author.new
```
Comment (vo-thanh-mulodo) [inline app/controllers/authors_controller.rb]: @tran-sang-mulodo 
This instance variable should be singular.

File: app/controllers/authors_controller.rb [hunk 2]
```diff
@@ -15,33 +15,31 @@ def show
...
-    respond_to do |format|
-      format.html
-    end
   end
 
   def create
     @authors = Author.create(name: post_params['name'], image: post_params['image'], description: post_params['description'])
```
Comment (vo-thanh-mulodo) [inline app/controllers/authors_controller.rb]: @tran-sang-mulodo 
Just Author.create(post_params) is enough

File: app/controllers/authors_controller.rb [hunk 3]
```diff
@@ -15,33 +15,31 @@ def show
...
+      if @authors.save
+        format.js { redirect_to authors_path, notice: 'Author has been created by ajax..'}
+      else
+        format.js
+      end
     end
   end
 
   def edit
     @authors = Author.find_by_id(params['id'])
```
Comment (vo-thanh-mulodo) [inline app/controllers/authors_controller.rb]: @tran-sang-mulodo 
Don't repeat it in many methods. You should do set it and call like method 'post_params'

File: app/controllers/authors_controller.rb [hunk 4]
```diff
@@ -1,4 +1,8 @@
 class AuthorsController < ApplicationController
+  before_action :set_author, only: [:show, :edit, :update, :destroy]
```
Comment (truong-dat-mulodo) [inline app/controllers/authors_controller.rb]: :show đâu cần thiết set_author

File: app/controllers/authors_controller.rb [hunk 5]
```diff
@@ -14,39 +18,34 @@ def show
   end
```
Comment (vo-thanh-mulodo) [inline app/controllers/authors_controller.rb]: @tran-sang-mulodo 
set_author has set '@authors' for show action So you don't need set it again. And respond to html is default so you need it too

**Pull request merged**